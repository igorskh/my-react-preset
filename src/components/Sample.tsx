import React, { useState } from "react";
import Child from "./Child";

import "./Sample.css";

type AppProps = {
    title: string,
    number: number
}

const Sample = ({ title, number }: AppProps): JSX.Element => {
    const [counter, setCounter] = useState<number>(number);

    function changeCounter(diff: number) {
        setCounter(p => p + diff);
    }

    return (
        <div id="block">
            <h1>{title}</h1>
            <h2>Count is {counter}</h2>

            <Child text={"test"} />

            <button onClick={() => changeCounter(1)}>+1</button>
            <button onClick={() => changeCounter(-1)}>-1</button>
        </div>
    );
};

export default Sample;
