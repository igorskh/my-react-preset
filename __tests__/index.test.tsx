import React from "react";
import * as renderer from "react-test-renderer";

import Sample from "../src/components/Sample";

it("SubjectToBeTested renders correctly", () => {
  const props = { title: "test", number: 42 };
  const tree = renderer
    .create(<Sample {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});