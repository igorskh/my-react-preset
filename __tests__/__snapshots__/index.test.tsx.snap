// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectToBeTested renders correctly 1`] = `
<div
  id="block"
>
  <h1>
    test
  </h1>
  <h2>
    Count is 
    42
  </h2>
  <div>
    <p>
      test
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
